### ISSUE TITLE
COPY TO  parquet with ZSTD produces broken column with 100k rows (~3GB uncompressed, 45MB compressed)

### ISSUE THREAD

**Issue by corneliusroemer** [duckdb/duckdb#2815]
#### What happens?
COPY TO  parquet with ZSTD produces broken column with 100k rows (~3GB uncompressed, 30MB compressed)

```python
duckdb.query("""
    SELECT sequence FROM 'duckseq.parquet'
    USING SAMPLE 10;
    """).fetchall()

RuntimeError                              Traceback (most recent call last)
~/code/duckdb-test/test.py in <module>
----> 47 duckdb.query("""
      48     SELECT sequence FROM 'duckseq.parquet'
      49     USING SAMPLE 10;
      50     """).fetchall()

RuntimeError: ZSTD Decompression failure
```

#### To Reproduce
```python
import duckdb
con = duckdb.connect("::memory::")
con.execute("""
    COPY (SELECT * FROM read_csv_auto('sequences.csv.gz', delim=',', header=True) )
    TO 'duckseq.parquet' 
    (FORMAT 'PARQUET', CODEC 'ZSTD');
    """)
```
File: https://transfer.sh/0ugKVN/sequences.csv.gz

#### Environment (please complete the following information):
 - OS: macOS 12.1 Intel
 - DuckDB Version: 0.3.1
 - DuckDB Client: Python

#### Possible root cause

I think I know what happens. When I observe htop, I see that the memory requirements balloon, going to 30% and more. Then suddenly memory goes to 0 while python command still runs. Only a few seconds later does Python complete, _without throwing an error_.

It looks like the error handling is broken. The process gets killed as OOM but Python never tells the end user about it nor cleans up the resulting file.

**Comment by Akaame** [comment]
@corneliusroemer Hello, how do you inspect the resulting parquet file and see that it has a broken column?

I get the same error during parquet read with additional error message:

```
Python(15948,0x11a174e00) malloc: can't allocate region
:***  (size=18446744072397123584, flags: 100) failed (error code=3)
Python(15948,0x11a174e00) malloc: *** set a breakpoint in malloc_error_break to debug
Traceback (most recent call last):
  File "/\path/to/oom.py", line 9, in <module>
    con.execute("""
RuntimeError: ZSTD Decompression failure
```

Why does it mean for sample clause to ve dimensionless?

```
SELECT sequence FROM 'duckseq.parquet'
    USING SAMPLE 10;
```

Should not this be

```
SELECT sequence FROM 'duckseq.parquet'
    USING SAMPLE 10%;
```

[Syntax](https://duckdb.org/docs/sql/query_syntax/sample)

**Comment by Mytherin** [comment]
I have pushed a fix to this problem in #2821. The problem is that the 100K rows exceed 2GB in uncompressed size, which exceed the uncompressed page size limit in Parquet files, as this is stored as an INT32 for reasons.
